src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/WriteSbrpUsageReport.cs

```diff
@@ -0,0 +1,195 @@
+                Version = version,
+                Name = projectName[..(projectName.Length - 1 - version.Length)],
+                Path = directory.FullName,
+            };
+
+            if (trackTfms)
+            {
+                XDocument xmlDoc = XDocument.Load(projectPath);
+                // Reference packages are generated using the TargetFrameworks property
+                // so there is no need to handle the TargetFramework property.
+                string[] tfms = xmlDoc.Element("Project")?
+                    .Elements("PropertyGroup")
+                    .Elements("TargetFrameworks")
+                    .FirstOrDefault()?.Value?.Split(';');
+
+                if (tfms == null || tfms.Length == 0)
+                {
+                    Log.LogError($"No TargetFrameworks were delected in {projectPath}.");
+                }
+
+                info.Tfms = new HashSet<string>(tfms);
+            }
+            else
+            {
+                info.Tfms = [];
+            }
+
+            _sbrpPackages.Add(info.Id, info);
+            Log.LogMessage($"Detected package: {info.Id}");
+        }
+    }
+
+    private void ScanProjectReferences()
+    {
+        EnumerationOptions options = new() { RecurseSubdirectories = true };
+
+        foreach (string projectJsonFile in Directory.GetFiles(SrcPath, "project.assets.json", options))
+        {
+            LockFile lockFile = new LockFileFormat().Read(projectJsonFile);
+            foreach (LockFileTargetLibrary lib in lockFile.Targets.SelectMany(t => t.Libraries))
+            {
+                if (!_sbrpPackages.TryGetValue($"{lib.Name}/{lib.Version}", out PackageInfo info))
+                {
+                    continue;
+                }
+
+                if (!info.References.TryGetValue(lockFile.Path, out HashSet<string> referencedTfms))
+                {
+                    referencedTfms = [];
+                    info.References.Add(lockFile.Path, referencedTfms);
+                }
+
+                IEnumerable<string> tfms = lib.CompileTimeAssemblies
+                    .Where(asm => asm.Path.StartsWith("lib") || asm.Path.StartsWith("ref"))
+                    .Select(asm => asm.Path.Split('/')[1]);
+                foreach (string tfm in tfms)
+                {
+                    referencedTfms.Add(tfm);
+                }
+            }
+        }
+    }
+
+    private class PackageInfo
+    {
+        public string Id => $"{Name}/{Version}";
+        public string Name { get; set; }
+        public string Version { get; set; }
+        public string Path { get; set; }
+        public HashSet<string> Tfms { get; set; }
+
+        // Dictionary of projects referencing the SBRP and the TFMs referenced by each project
+        public Dictionary<string, HashSet<string>> References { get; } = [];
+    }
+
+    private class Report
+    {
+        public PackageInfo[] Sbrps { get; set; }
+        public string[] UnreferencedSbrps { get; set; }
+    }
```

Comment:
```suggestion
    private record Report(PackageInfo[] Sbrps, string[] UnreferencedSbrps);
```

Reply:
You will also need to update the instantiation of the `Report` record above in L55.